bluepysnap/network.py

```diff
@@ -164,6 +164,15 @@ def get(self, group=None, properties=None):
             # indices from Population and get functions are different so I cannot
             # use a dataframe equal directly and properties have different types so cannot use a
             # multi dim numpy array
-            for prop in pop_properties:
-                res.loc[global_pop_ids.index, prop] = pop.get(pop_ids, prop).to_numpy()
+            if len(global_pop_ids) < len(res):
+                for prop in pop_properties:
+                    res.loc[global_pop_ids.index, prop] = pop.get(pop_ids, prop).to_numpy()
+            else:
+                # Dirty hack to avoid:
+                # FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt
+                # to set the values inplace instead of always setting a new array.
+                # To retain the old behavior, use either `df[df.columns[i]] = newvals` or,
+                # if columns are non-unique, `df.isetitem(i, newvals)`
+                for prop in pop_properties:
+                    res[prop] = pop.get(pop_ids, prop).to_numpy()
```

Comment:
I think @GianlucaFicarelli's way is cleaner; that way other future warnings are impacted.

Is this really a dirty hack?

Reply:
tbh using `with warnings` as a context manager with a regexp doesn't look too bad because it won't affect other lines of code.
However, I'm still thinking if there is a different way.... (moving the PR to draft)

Reply:
According to a small test I did, all the columns are objects (`dtype('O')`), and if the insert is done in-place, the dtype is kept. Otherwise, the dtype changes to that of the inserted values. What I don't understand, and it might be that I'm missing something here, is that even in `pandas==1.4.0`, the behavior is
```
In [1]: import pandas as pd

In [2]: df=pd.DataFrame(columns=list('abcd'), index=list(range(20)))

In [3]: df['c'] = range(0,40,2)

In [4]: df['c'].dtype
Out[4]: dtype('int64')

In [5]: df.loc[range(20), 'b'] = range(0,40,2)

In [6]: df['b'].dtype
Out[6]: dtype('O')
```
So to me, it seems that the `df.loc[...] = array` has done the insertion in-place already. 🤔

Reply:
I see. Thanks for the clarification! 🤘 Now to the question: do we want the previous behaviour or just suppress the warning. Going with the previous would be bug-safer but I don't really see how either would cause bugs (famous last words here).

Also, I don't think we would get a performance increase significant enough if the insertion was done in place, so I'm starting to lean towards your "dirty hack" (it's not that bad, really) to be on the safe side.

Reply:
Even more confusing, the warning mentions `df.iloc[:, i]` and not `df.loc[df.index, col]`.

This code for example always produces a warning, even with a dataframe with a Int64Index, and when the right value is a range (not numpy array):
```
In [1]: import pandas as pd

In [2]: df=pd.DataFrame(columns=list('abcd'), index=list(range(20)))

In [3]: df.loc[:, 'b'] = range(0,40,2)
<ipython-input-3-dde28b2bf3b1>:1: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df.loc[:, 'b'] = range(0,40,2)

In [4]: df['b'].dtype
Out[4]: dtype('int64')
```

Reply:
Well, ain't that beautiful! 😮 
I'm also not against holding off on `1.5.0` with the hope that these would be clarified/mitigated in a minor release.

Reply:
I changed the code to create the DataFrame as a concatenation, so that there are not warnings, and the performance looks better too.
I tested the `get()` with a circuit with 4M+ nodes and a single population and the time needed to load all the nodes went from 38-39 to 6-7 seconds (around 5 times faster).
If only a subset of nodes is loaded, the difference seems lower but there is still an improvement.
I didn't test the performance with a circuit with multiple populations, but I can try it if we have one.

Note that I updated the tests to check the full DataFrame, and the new tests would fail with the old code because some of the dtypes are different (they are now int or float instead of object)

Circuit used:
```
from bluepysnap import Circuit
circuit = Circuit("/gpfs/bbp.cscs.ch/project/proj83/circuits/Bio_M/20200805/sonata/circuit_config.json")
%time circuit.nodes.get()
```